lib/node_modules/@stdlib/repl/lib/output_stream.js

```diff
@@ -0,0 +1,525 @@
+
+// Define the number of rows which are reserved for displaying the paging UI (including the input prompt):
+var RESERVED_PAGING_ROWS = 4;
+
+
+// FUNCTIONS //
+
+/**
+* Returns a list of indices corresponding to line segments.
+*
+* @private
+* @param {string} str - input string
+* @returns {Array<integer>} list of indices
+*
+* @example
+* var str = 'foo\nbar\nbeep\nboop\n';
+*
+* var idx = findLineSegments( str );
+* // returns [ 0, 4, 8, 13, 18 ]
+*/
+function findLineSegments( str ) {
+	var lines;
+	var out;
+	var i;
+
+	lines = str.split( RE_EOL ); // FIXME: this approach is not robust for three reasons: (1) line breaks may include carriage returns, not just line feeds (so the `+1` may not be appropriate), (2) it does not consider line feeds which are escaped (e.g., `\\\n`), and (3) assumes that line breaks map directly to displayed lines (while true most of the line for text which has been wrapped to 80 chars, such as repl.txt files, this may not be true for all outputs, some of which may spill across multiple lines)
+	out = [ 0 ];
+	for ( i = 0; i < lines.length-1; i++ ) {
+		out.push( out[ i ] + lines[ i ].length + 1 );
+	}
+	return out;
+}
+
+
+// MAIN //
+
+/**
+* REPL output stream.
+*
+* @private
+* @constructor
+* @param {REPL} repl - REPL instance
+* @param {boolean} autoPage - boolean indicating whether auto-paging should be initially enabled
+* @returns {OutputStream} transform stream
+*/
+function OutputStream( repl, autoPage ) {
+	if ( !( this instanceof OutputStream ) ) {
+		return new OutputStream( repl, autoPage );
+	}
+	debug( 'Creating an output stream...' );
+	Transform.call( this, {
+		'encoding': 'utf8',
+		'decodeString': false
+	});
+
+	// Cache a reference to the REPL instance:
+	this._repl = repl;
+
+	// Initialize a flag indicating whether "paging" is enabled:
+	this._paging = autoPage;
+
+	// Initialize a flag indicating whether "paging" is currently active:
+	this._isPaging = false;
+
+	// Initialize a buffer for caching streamed chunks:
+	this._buffer = '';
+
+	// Initialize a buffer for storing line breaks:
+	this._indices = [];
+
+	// Initialize a variable for tracking the current scroll position:
+	this._idx = -1;
+
+	// Initialize a flag indicating whether the stream has been destroyed:
+	this._destroyed = false;
+
+	return this;
+}
+
+/**
+* Inherit from the `Transform` prototype.
+*/
+inherit( OutputStream, Transform );
+
+/**
+* Implements the `_transform` method.
+*
+* @private
+* @name _transform
+* @memberof OutputStream.prototype
+* @type {Function}
+* @param {(Buffer|string)} chunk - streamed chunk
+* @param {string} encoding - Buffer encoding
+* @param {Callback} clbk - callback to invoke after transformed the streamed chunk
+* @returns {void}
+*/
+setNonEnumerableReadOnly( OutputStream.prototype, '_transform', function transform( chunk, encoding, clbk ) {
+	var indices;
+
+	// debug( 'Received a new chunk. Chunk: %s. Encoding: %s.', chunk.toString(), encoding );
```

Comment:
@kgryte Found a commented out debug log, is it intentional?

Reply:
It is intentional. I should add a comment. Namely, it is useful to have here, but you don't want this always, as it can create a lot of noise when debugging.